File: on_audio_query/android/src/main/kotlin/com/lucasjosino/on_audio_query/controllers/MethodController.kt
package com.lucasjosino.on_audio_query.controller

import android.content.Context
import com.lucasjosino.on_audio_query.consts.Method
import com.lucasjosino.on_audio_query.queries.*
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class MethodController(
    private val context: Context,
    private val call: MethodCall,
    private val result: MethodChannel.Result
) {

    //
    fun find() {
        when (call.method) {
            //Query methods
            Method.QUERY_AUDIOS -> AudioQuery().querySongs(context, result, call)
            Method.QUERY_ALBUMS -> AlbumQuery().queryAlbums(context, result, call)
            Method.QUERY_ARTISTS -> ArtistQuery().queryArtists(context, result, call)
            Method.QUERY_PLAYLISTS -> PlaylistQuery().queryPlaylists(context, result, call)
            Method.QUERY_GENRES -> GenreQuery().queryGenres(context, result, call)
            Method.QUERY_ARTWORK -> ArtworkQuery().queryArtwork(context, result, call)
            Method.QUERY_AUDIOS_FROM -> AudioFromQuery().querySongsFrom(context, result, call)
            Method.QUERY_WITH_FILTERS -> WithFiltersQuery().queryWithFilters(context, result, call)
            Method.QUERY_ALL_PATHS -> AllPathQuery().queryAllPath(context, result)
            //Playlists methods
            Method.CREATE_PLAYLIST -> PlaylistController().createPlaylist(context, result, call)
            Method.REMOVE_PLAYLIST -> PlaylistController().removePlaylist(context, result, call)
            Method.ADD_TO_PLAYLIST -> PlaylistController().addToPlaylist(context, result, call)
            Method.REMOVE_FROM_PLAYLIST -> PlaylistController().removeFromPlaylist(
                context,
                result,
                call
            )
            Method.RENAME_PLAYLIST -> PlaylistController().renamePlaylist(context, result, call)
            Method.MOVE_ITEM_TO -> PlaylistController().moveItemTo(context, result, call)
            else -> result.notImplemented()
        }
    }
}

File: packages/on_audio_query_ios/ios/Classes/controllers/MethodController.swift
import Flutter

public class MethodController {
    public func find() {
        let call = try! PluginProvider.call()
        let result = try! PluginProvider.result()

        // All necessary method to this plugin support both platforms, only playlists
        // are limited when using [IOS].
        switch call.method {
        case Method.QUERY_AUDIOS:
            AudioQuery().querySongs()
        case Method.QUERY_ALBUMS:
            AlbumQuery().queryAlbums()
        case Method.QUERY_ARTISTS:
            ArtistQuery().queryArtists()
        case Method.QUERY_GENRES:
            GenreQuery().queryGenres()
        case Method.QUERY_PLAYLISTS:
            PlaylistQuery().queryPlaylists()
        case Method.QUERY_AUDIOS_FROM:
            AudioFromQuery().queryAudiosFrom()
        case Method.QUERY_WITH_FILTERS:
            WithFiltersQuery().queryWithFilters()
        case Method.QUERY_ARTWORK:
            ArtworkQuery().queryArtwork()
        // The playlist for [IOS] is completely limited, the developer can only:
        //   * Create playlist
        //   * Add item to playlist (Unsuported, for now)
        //
        // Missing methods:
        //   * Rename playlist
        //   * Remove playlist
        //   * Remove item from playlist
        //   * Move item inside playlist
        case Method.CREATE_PLAYLIST:
            PlaylistController().createPlaylist()
        case Method.ADD_TO_PLAYLIST:
            PlaylistController().addToPlaylist()
        default:
            // All non suported methods will throw this error.
            result(FlutterMethodNotImplemented)
        }
    }
}
